apps/admin-gui/src/app/vos/pages/group-detail-page/group-subgroups/group-subgroups.component.html

```diff
@@ -25,6 +25,37 @@ <h1 class="page-subtitle">{{'GROUP_DETAIL.SUBGROUPS.TITLE' | translate}}</h1>
       {{'GROUP_DETAIL.SUBGROUPS.DELETE_GROUP' | translate}}
     </button>
   </span>
+
+  <mat-form-field class="me-2 fix-select">
+    <mat-label>{{'GROUP_DETAIL.SUBGROUPS.FILTER_ROLES' | translate}}</mat-label>
+    <mat-select (closed)="changeRoles()" [formControl]="roles" multiple>
+      <mat-select-trigger>
+        {{displaySelectedRoles() | groupRole}}
+      </mat-select-trigger>
+      <mat-option
+        class="vo-theme"
+        *ngFor="let role of rolesList"
+        [value]="role"
+        >{{role | groupRole}}</mat-option
+      >
+    </mat-select>
+  </mat-form-field>
+
+  <mat-form-field class="filter-field me-2 fix-select">
+    <mat-label>{{'GROUP_DETAIL.SUBGROUPS.FILTER_ROLE_TYPES' | translate}}</mat-label>
+    <mat-select (closed)="changeRoleTypes()" [formControl]="roleTypes" multiple>
```

Comment:
```suggestion
    <mat-select (selectionChange)="changeRoleTypes()" [formControl]="roleTypes" multiple>
```
Only refresh page when the chosen roles change. This should do the job I believe